### ISSUE TITLE
Pagmo installation on windows

### ISSUE THREAD

**Issue by ShavaraniMahdi** [esa/pagmo2#258]
Hi,

I was trying to install Pagmo according to the instructions given in the documentation. I encountered some problems that took me a long time to resolve and I thought it is a good idea to share my experience in order to enrich the documentation for future references.
When using CMake in windows for configuring Pagmo it should be noticed that:
1. The CMake should be run as administrator. 
2. Boost library should be added to the entry list.
3. CMD should be run as administrator.
4. The command "make install" does not work in windows, instead, you should use "CMake make install"

I hope these notes would be helpful to other users like me that are not experts.

Best,
Mahdi

**Comment by bluescarni** [comment]
Running cmake/cmd as an admin is not a good idea, and it should not be necessary. We regularly produce pagmo/pygmo binary packages on Windows and we never have to run anything with special privileges. What kind of errors are you getting?

Indeed ``make install`` won't work on Windows, unless you have some sort of POSIX emulation layer or you are using MinGW. The installation instructions from source could use some love, PRs are welcome as usual :)

**Comment by ShavaraniMahdi** [comment]
well, the installation process was a little bit hard for me. As some steps were not mentioned assuming that all the users are experts.
for one if I did not add boost manually cmake would have not configured the project. or when running the command camke make install, there was an error "the source directory does not exist".

**Comment by bluescarni** [comment]
@ShavaraniMahdi in general we tend to assume regular users don't want to compile pagmo/pygmo from source, and we direct them to use the binary packages from ``conda`` instead. Perhaps this is not clear enough in the documentation? Or are there specific reasons why you needed to compile from source?

(for the record, we agree that the introductory documentation is lacking, we have various improvements planned for the near future)

**Comment by ShavaraniMahdi** [comment]
Yes, it is not really clear and I just followed the instructions and conda was only suggested as an option. 
I tried everything I could and finally installed the package through visual studio. Then I ran the cmd command in the documentation for quick-start example but the system wouldn't find the pagmo and boost. I added the libraries by the following command
g++ -O2 -DNDEBUG -std=c++11 test.cpp -I ~pagmo\include -I ~\boost
it started compiling but gave me many errors one of which is the lack of header <mutex>.

I'm really looking forward to the improvements in documentation.

**Comment by Nenad7955** [comment]
i'm also on windows, and for some reason nothing won't start because a<pagmo/config.hpp> is missing, in files like serialization, island.

**Comment by darioizzo** [comment]
@Nenad7955 Try to be more specific when reporting a problem, otherwise its really hard to help.
What do you mean "nothin won't start"? Are you trying to compile some code after having installed pagmo headers? If so, how did you instal the headers? conda? from source? Do you have a minimal example of code that will not compile? What compiler are you using? etc..
